lib/uri.js

```diff
@@ -90,6 +93,10 @@ URI.prototype.toString = function(options) {
                 // Omit optional segment & return
                 return uriStr;
             }
+        } else if (this._pathMetadata
+                && this._pathMetadata[i]
+                && this._pathMetadata[i] === '+') {
+            uriStr += '/' + utils.encodeReserved(segment);
```

Comment:
I think you should be able to simply skip the encoding here. http://tools.ietf.org/html/rfc6570#page-22 says:

> For each defined variable in the variable-list, perform variable expansion, as defined in Section 3.2.1, with the allowed characters being those in the set (unreserved / reserved / pct-encoded).  If more than one variable has a defined value, append a comma (",") to the result string as a separator between variable expansions.